docs/docsite/rst/dev_guide/developer_labs.rst

```diff
@@ -0,0 +1,136 @@
+Developer Labs
+==============
+
+.. contents:: Topics
+
+Ansible has a lot of documentation and is increasingly building out development docs, but not everyone learns the same way. This page provides self-guided labs which are intended to teach new Ansible developers the ins and outs of writing code and working with the github PR process and shippable CI system. It is expected that the developer does a commit+push and waits for tests to run after each step in the process to learn the behavior of shippable tests and ansibot. You are expected to see a lot of errors and failures through this process. Please use the docs, the sourcecode, the community and the Ansible core team (in that order) if you find yourself stuck at any point.
+
+.. _level_1:
+
+Level 1
+-------
+
+.. _module:
+
+Module
+++++++
+
+Build a module and tests from scratch by starting with a broken python script.
+
+.. _module_hello_world:
+
+module from scratch
+```````````````````
+
+1. fork ansible
+2. checkout the fork
+3. create a feature branch named FILAMENT_MODULE
+4. create a new module file lib/ansible/modules/system/filament.py
+5. put "hello world" in filament.py
+6. git add and commit the new file
+7. push the commit to the feature branch
+8. open the branch on github.com and send it as a pullrequest to ansible/ansible
```

Comment:
actually, with shippable enabled, you dont need PR technically speaking, but good way to show how it would work against main repo